docs/src/content/en/docs/workflows-vnext/inngest-workflow.mdx

```diff
@@ -7,59 +7,84 @@ description: "Inngest workflow allows you to run Mastra vNext workflows with Inn
 
 [Inngest](https://www.inngest.com/docs) is a developer platform for building and running background workflows, without managing infrastructure.
 
-## Setup
+## How Inngest Works with Mastra
 
-```sh
-npm install @mastra/inngest @mastra/core @mastra/deployer @hono/node-server
-```
+Inngest provides an event delivery and orchestration layer for running Mastra vNext workflows in response to events. The integration connects Inngest’s event system to Mastra’s workflow engine, enabling event-driven execution, step-level monitoring, and suspend/resume functionality.
 
-### Local Development Environment
+### Integration Architecture
 
-Inngest offers two methods for local development:
+- **Workflow Registration:**  
+  Workflows are defined using the `createWorkflow` and `createStep` helpers from `@mastra/inngest`. These helpers produce workflow and step objects compatible with both Mastra and Inngest.
 
-#### Option 1: Using Docker
+- **Event Handling:**  
+  Each workflow is registered as an Inngest function. When an event matching the workflow’s trigger is received, Inngest invokes the workflow handler. The handler starts or resumes a workflow run using the event payload as input.
 
-Run Inngest locally using Docker on port 8288, configured to listen for events on port 3000:
+- **Run Management:**  
+  Workflow runs are managed by the `InngestRun` class. This class sends events to Inngest to start or resume runs, polls for run status, and retrieves results. Workflow state is persisted using Mastra’s storage system.
 
-```sh
-docker run --rm -p 8288:8288 \
-  inngest/inngest \
-  inngest dev -u http://host.docker.internal:3000/inngest/api
-```
+- **Suspend and Resume:**  
+  If a workflow step suspends (for example, waiting for external input), the run is paused. To resume, a new event is sent to Inngest with the required resume data and step information. The workflow continues from the suspended step.
+
+- **Real-Time Monitoring:**  
+  The `watch` method on a workflow run subscribes to Inngest’s realtime API. This provides step-level execution updates, including status changes and outputs, which can be used for dashboards or debugging.
```

Comment:
Worth describing here that we rely on the realtime middleware from inngest for this and use their publish-subscribe paradigm for it.